boruta/script.whizzml

```diff
@@ -114,9 +120,47 @@
                                                             IMPORTANCE_DEC
                                                             to-name))
             (log-info "Undecided: " (select-by-importance f-rank 0 to-name))
-            (log-info "remaining fields: " (map to-name (keys fields)))
-            (log-info "------------------------------------------------")
-            (recur f-rank fields sh-ds-id (+ runs 1)))))))
+            (log-info "Remaining fields: " (map to-name (keys fields)))
+            (log-info "---------------------------------------------------------")
+            (recur f-rank fields sh-ds-id max-shadows f-importances (+ runs 1)))))))
+
+
+;; median-importances
+;; Obtains median importance of a field from a list of maps (field -> importance)
+;;
+;; Inputs:
+;;   f-importances: (list) List of maps from field ID to imporance flag
+;;                         from each iteration
+;;
+;; Output: (number) Median importance
+;;
+(define (median-importances f-importances field)
+ (median (map (lambda (x) (x field 0)) f-importances)))
+
+;; median
+;; Returns median value of list of numbers
+;; If the list contains an even number of elements,
+;; it returns the average of both central elements
+;;
+;; Inputs:
+;;   myList: (list) List of numbers
+;;
+;; Output: (number) Median importance
+;;
+(define (median mylist)(
+let(
+    mylist (sort mylist)
+    med-item (floor (/ (count mylist) 2))
+    prev-item  (- med-item 1)
+    prev-item (if (> prev-item 0) prev-item 0)
+    value (if (empty? mylist)
+           0
+          (if(odd? (count mylist))
+              (mylist med-item)
+              (/ (+ (mylist med-item)  (mylist prev-item) )2)))
+    )value
+))
```

Comment:
indentation is off... something like:
```
(define (foo bar)
  (let (mylist ...
       med-item --)
    value))
```

please see at the other functions and code for more examples